File: app/src/main/java/com/suhang/networkmvp/adapter/HomeRvAdapter.java
package com.suhang.networkmvp.adapter;

import android.view.View;
import android.view.ViewGroup;

import com.suhang.networkmvp.R;
import com.suhang.networkmvp.adapter.viewholder.BaseViewHolder;
import com.suhang.networkmvp.binding.data.AdapterHomeRvData;
import com.suhang.networkmvp.binding.data.BaseData;
import com.suhang.networkmvp.databinding.ItemHomeBinding;
import com.suhang.networkmvp.domain.HomeBean;

import javax.inject.Inject;

/**
 * Created by 苏杭 on 2017/1/24 16:51.
 */

public class HomeRvAdapter extends BaseRvAdapter<HomeRvAdapter.MyViewHolder,HomeBean.ListEntity>  {

    @Inject
    public HomeRvAdapter() {
    }

    @Override
    protected BaseData getBindingData() {
        return new AdapterHomeRvData();
    }


    @Override
    public void onBindHolder(MyViewHolder holder, HomeBean.ListEntity listEntity) {
        holder.mBinding.getData();
        holder.mBinding.tv.setText(listEntity.getNick());
        holder.mBinding.getRoot().setBackgroundColor(0xffffffff);
    }


    @Override
    public MyViewHolder onCreateHolder(ViewGroup parent, int viewType) {
        return new MyViewHolder(View.inflate(getContext(),R.layout.item_home,null));
    }

    @Override
    public int getPageSize() {
        return 10;
    }

    public class MyViewHolder extends BaseViewHolder<ItemHomeBinding> {
        public MyViewHolder(View view) {
            super(view);
        }
    }
}

File: app/src/main/java/com/suhang/networkmvp/adapter/HomeRvAdapter.kt
package com.suhang.networkmvp.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup

import com.suhang.networkmvp.R
import com.suhang.networkmvp.constants.setAdapterTag
import com.suhang.networkmvp.domain.HistoryBean
import com.suhang.networkmvp.mvp.result.EventResult
import javax.inject.Inject
import kotlinx.android.synthetic.main.item_home.view.*

/**
 * Created by 苏杭 on 2017/1/24 16:51.
 */

class HomeRvAdapter @Inject
constructor(context: Context) : BaseRvAdapter<HomeRvAdapter.MyViewHolder, HistoryBean.ListEntity>(context) {
    override fun onBindHolder(holder: MyViewHolder, v: HistoryBean.ListEntity) {
        holder.itemView.tv.text = v.nick
        holder.itemView.setBackgroundColor(0xffffffff.toInt())
        holder.itemView.setOnClickListener {
            it.setAdapterTag(Message(holder.adapterPosition, v.uid))
            manager.post(EventResult(it))
        }
    }

    data class Message(var position: Int, var luid: String)


    override fun onCreateHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        return MyViewHolder(bind(R.layout.item_home))
    }

    class MyViewHolder(view: View) : RecyclerView.ViewHolder(view)
}
